Panic executing `okteto destroy --remote` when there is no okteto manifest

**Describe the bug**

I was able to deploy my repository using `okteto deploy --remote` but when I executed `okteto destroy --remote` I got a panic

```
╰─ okteto destroy --remote
 i  Using test-nacho @ okteto.mario18rc.dev.okteto.net as context
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2c43057]

goroutine 1 [running]:
github.com/okteto/okteto/cmd/destroy.newRemoteDestroyer(0xc001c74000)
	github.com/okteto/okteto/cmd/destroy/remote.go:104 +0x217
github.com/okteto/okteto/cmd/destroy.(*destroyCommand).getDestroyer(0xc001a3fbc0, {0x3b54878, 0xc0001a0000}, 0xc0001e5880)
	github.com/okteto/okteto/cmd/destroy/destroy.go:269 +0x345
github.com/okteto/okteto/cmd/destroy.Destroy.func1(0xc001862f00?, {0xc00184c840?, 0x0?, 0x1?})
	github.com/okteto/okteto/cmd/destroy/destroy.go:193 +0x990
github.com/spf13/cobra.(*Command).execute(0xc001862f00, {0xc00184c830, 0x1, 0x1})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc001848780)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	github.com/okteto/okteto/main.go:148 +0xa25
```

I think the error might come from the fact I'm not having an `okteto.yml` file in my root folder. Instead, I have a `k8s.yml` file.

The repository is https://github.com/ifbyol/test-pipeline and the branch `destroy-custom-manifest`

**To Reproduce**
Steps to reproduce the behavior:

1. Clone the repository https://github.com/ifbyol/test-pipeline
2. Change to the branch `destroy-custom-manifest`
3. Execute `okteto deploy --remote`. You should see how the application is successfully deployed
4. Execute `okteto destroy --remote`. You should see the panic error

**Expected behavior**

As I was able to deploy the application correctly, I would expect to be able to destroy it successfully. If something is wrong, I would expect a clear error message instead of a stack trace

**Desktop (please complete the following information):**

- Version: `2.15.3` and master

**Additional context**
N/A

--- a/cmd/deploy/deploy.go
+++ b/cmd/deploy/deploy.go
@@ -49,7 +49,7 @@ const (
 	succesfullyDeployedmsg = "Development environment '%s' successfully deployed"
 )
 
-// Options options for deploy command
+// Options represents options for deploy command
 type Options struct {
 	// ManifestPathFlag is the option -f as introduced by the user when executing this command.
 	// This is stored at the configmap as filename to redeploy from the ui.

--- a/cmd/destroy/destroy.go
+++ b/cmd/destroy/destroy.go
@@ -235,14 +235,11 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 	} else {
 		manifest, err := model.GetManifestV2(opts.ManifestPath)
 		if err != nil {
-			// error will only be available when there is no valid manifest found
-			// hence, we need not to destroy because the manifest would never be deployed
-			// Return the error message
+			// Log error message but application can still be deleted
 			oktetoLog.Infof("could not find manifest file to be executed: %s", err)
 			manifest = &model.Manifest{
 				Destroy: &model.DestroyInfo{},
 			}
-			return nil, err
 		}
 
 		isRemote := utils.LoadBoolean(constants.OktetoDeployRemote)

--- a/cmd/destroy/destroy.go
+++ b/cmd/destroy/destroy.go
@@ -270,7 +270,7 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 		// if the command is executed remotely and destroy section is nil, this will cause a panic
 		// Hence consider the command locally for successful execution.
 		// This will make sure all the resources are deleted.
-		if runInRemote == true && manifest.Destroy == nil {
+		if manifest.Destroy == nil {
 			runInRemote = false
 		}
 

--- a/cmd/destroy/destroy.go
+++ b/cmd/destroy/destroy.go
@@ -16,10 +16,6 @@ package destroy
 import (
 	"context"
 	"fmt"
-	"os"
-	"path/filepath"
-	"time"
-
 	contextCMD "github.com/okteto/okteto/cmd/context"
 	"github.com/okteto/okteto/cmd/utils"
 	"github.com/okteto/okteto/cmd/utils/executor"
@@ -27,7 +23,6 @@ import (
 	"github.com/okteto/okteto/pkg/constants"
 	"github.com/okteto/okteto/pkg/devenvironment"
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
-
 	"github.com/okteto/okteto/pkg/k8s/kubeconfig"
 	"github.com/okteto/okteto/pkg/k8s/namespaces"
 	"github.com/okteto/okteto/pkg/k8s/secrets"
@@ -37,6 +32,9 @@ import (
 	oktetoPath "github.com/okteto/okteto/pkg/path"
 	"github.com/spf13/cobra"
 	v1 "k8s.io/api/core/v1"
+	"os"
+	"path/filepath"
+	"time"
 )
 
 const (
@@ -55,7 +53,7 @@ type secretHandler interface {
 	List(ctx context.Context, ns, labelSelector string) ([]v1.Secret, error)
 }
 
-// Options destroy commands options
+// Options represents the options for destroy command
 type Options struct {
 	// ManifestPathFlag is the option -f as introduced by the user when executing this command.
 	// This is stored at the configmap as filename to redeploy from the ui.
@@ -220,15 +218,15 @@ func getTempKubeConfigFile(name string) string {
 
 func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (destroyInterface, error) {
 	var (
-		deployer destroyInterface
-		err      error
+		destroyer destroyInterface
+		err       error
 	)
 
 	if opts.DestroyAll {
 		if !okteto.Context().IsOkteto {
 			return nil, oktetoErrors.ErrContextIsNotOktetoCluster
 		}
-		deployer, err = newLocalDestroyerAll(dc.k8sClientProvider, dc.executor, dc.nsDestroyer, dc.oktetoClient)
+		destroyer, err = newLocalDestroyerAll(dc.k8sClientProvider, dc.executor, dc.nsDestroyer, dc.oktetoClient)
 		if err != nil {
 			return nil, err
 		}
@@ -237,11 +235,14 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 	} else {
 		manifest, err := model.GetManifestV2(opts.ManifestPath)
 		if err != nil {
-			// Log error message but application can still be deleted
+			// error will only be available when there is no valid manifest found
+			// hence, we need not to destroy because the manifest would never be deployed
+			// Return the error message
 			oktetoLog.Infof("could not find manifest file to be executed: %s", err)
 			manifest = &model.Manifest{
 				Destroy: &model.DestroyInfo{},
 			}
+			return nil, err
 		}
 
 		isRemote := utils.LoadBoolean(constants.OktetoDeployRemote)
@@ -265,8 +266,16 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 		}
 		runInRemote := !isRemote && (destroyImage != "" || opts.RunInRemote)
 
+		// destroy section should not be nil whether the command is executed remotely or locally
+		// if the command is executed remotely and destroy section is nil, this will cause a panic
+		// Hence consider the command locally for successful execution.
+		// This will make sure all the resources are deleted.
+		if runInRemote == true && manifest.Destroy == nil {
+			runInRemote = false
+		}
+
 		if runInRemote {
-			deployer = newRemoteDestroyer(manifest)
+			destroyer = newRemoteDestroyer(manifest)
 			oktetoLog.Info("Destroying remotely...")
 		} else {
 			destroyerAll, err := newLocalDestroyerAll(dc.k8sClientProvider, dc.executor, dc.nsDestroyer, dc.oktetoClient)
@@ -274,10 +283,10 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 				return nil, err
 			}
 
-			deployer = newLocalDestroyer(manifest, destroyerAll)
+			destroyer = newLocalDestroyer(manifest, destroyerAll)
 			oktetoLog.Info("Destroying locally...")
 		}
 	}
 
-	return deployer, nil
+	return destroyer, nil
 }

--- a/cmd/destroy/destroy.go
+++ b/cmd/destroy/destroy.go
@@ -266,14 +266,6 @@ func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (dest
 		}
 		runInRemote := !isRemote && (destroyImage != "" || opts.RunInRemote)
 
-		// destroy section should not be nil whether the command is executed remotely or locally
-		// if the command is executed remotely and destroy section is nil, this will cause a panic
-		// Hence consider the command locally for successful execution.
-		// This will make sure all the resources are deleted.
-		if manifest.Destroy == nil {
-			runInRemote = false
-		}
-
 		if runInRemote {
 			destroyer = newRemoteDestroyer(manifest)
 			oktetoLog.Info("Destroying remotely...")

--- a/cmd/destroy/remote.go
+++ b/cmd/destroy/remote.go
@@ -104,9 +104,12 @@ type remoteDestroyCommand struct {
 func newRemoteDestroyer(manifest *model.Manifest) *remoteDestroyCommand {
 	fs := afero.NewOsFs()
 	builder := remoteBuild.NewBuilderFromScratch()
+	if manifest.Destroy == nil {
+		manifest.Destroy = &model.DestroyInfo{}
+	}
 	return &remoteDestroyCommand{
 		builder:              builder,
-		destroyImage:         "",
+		destroyImage:         manifest.Destroy.Image,
 		fs:                   fs,
 		workingDirectoryCtrl: filesystem.NewOsWorkingDirectoryCtrl(),
 		temporalCtrl:         filesystem.NewTemporalDirectoryCtrl(fs),

--- a/cmd/destroy/remote.go
+++ b/cmd/destroy/remote.go
@@ -106,7 +106,7 @@ func newRemoteDestroyer(manifest *model.Manifest) *remoteDestroyCommand {
 	builder := remoteBuild.NewBuilderFromScratch()
 	return &remoteDestroyCommand{
 		builder:              builder,
-		destroyImage:         manifest.Destroy.Image,
+		destroyImage:         "",
 		fs:                   fs,
 		workingDirectoryCtrl: filesystem.NewOsWorkingDirectoryCtrl(),
 		temporalCtrl:         filesystem.NewTemporalDirectoryCtrl(fs),

--- a/cmd/destroy/remote.go
+++ b/cmd/destroy/remote.go
@@ -1,3 +1,16 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package destroy
 
 import (

--- a/pkg/model/manifest.go
+++ b/pkg/model/manifest.go
@@ -524,14 +524,14 @@ func GetInferredManifest(cwd string) (*Manifest, error) {
 		}
 		oktetoLog.AddToBuffer(oktetoLog.InfoLevel, "Found helm chart on %s", chartPath)
 		tags := inferHelmTags(chartPath)
-		command := fmt.Sprintf("helm upgrade --install ${%s} %s %s", constants.OktetoAutodiscoveryReleaseName, chartPath, tags)
+		deployHelm := fmt.Sprintf("helm upgrade --install ${%s} %s %s", constants.OktetoAutodiscoveryReleaseName, chartPath, tags)
 		chartManifest := &Manifest{
 			Type: ChartType,
 			Deploy: &DeployInfo{
 				Commands: []DeployCommand{
 					{
-						Name:    command,
-						Command: command,
+						Name:    deployHelm,
+						Command: deployHelm,
 					},
 				},
 			},

--- a/pkg/model/manifest_test.go
+++ b/pkg/model/manifest_test.go
@@ -1297,3 +1297,78 @@ func Test_Manifest_HasBuildSection(t *testing.T) {
 
 	}
 }
+
+func Test_getInferredManifestFromK8sManifestFile(t *testing.T) {
+	wd := t.TempDir()
+	fullpath := filepath.Join(wd, "k8s.yml")
+	f, err := os.Create(fullpath)
+	assert.NoError(t, err)
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatalf("Error closing file %s: %s", fullpath, err)
+		}
+	}()
+	_, err = GetInferredManifest(wd)
+	assert.NoError(t, err)
+}
+
+func Test_getInferredManifestFromK8sManifestFolder(t *testing.T) {
+	wd := t.TempDir()
+	fullpath := filepath.Join(wd, "manifests")
+	assert.NoError(t, os.MkdirAll(filepath.Dir(fullpath), 0750))
+	f, err := os.Create(fullpath)
+	assert.NoError(t, err)
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatalf("Error closing file %s: %s", fullpath, err)
+		}
+	}()
+
+	_, err = GetInferredManifest(wd)
+	assert.NoError(t, err)
+}
+
+func Test_getInferredManifestFromHelmPath(t *testing.T) {
+	var tests = []struct {
+		name          string
+		filesToCreate []string
+		expected      string
+	}{
+		{
+			name:          "chart folder exists on wd",
+			filesToCreate: []string{filepath.Join("chart", "Chart.yaml")},
+			expected:      "charts",
+		},
+		{
+			name:          "chart folder inside helm folder exists on wd",
+			filesToCreate: []string{filepath.Join("helm", "charts", "Chart.yaml")},
+			expected:      filepath.Join("helm", "charts"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wd := t.TempDir()
+			for _, fileToCreate := range tt.filesToCreate {
+				fullpath := filepath.Join(wd, fileToCreate)
+				assert.NoError(t, os.MkdirAll(filepath.Dir(fullpath), 0750))
+				f, err := os.Create(fullpath)
+				assert.NoError(t, err)
+				defer func() {
+					if err := f.Close(); err != nil {
+						t.Fatalf("Error closing file %s: %s", fullpath, err)
+					}
+				}()
+			}
+			_, err := GetInferredManifest(wd)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func Test_getInferredManifestWhenNoManifestExist(t *testing.T) {
+	wd := t.TempDir()
+	result, err := GetInferredManifest(wd)
+	assert.Empty(t, result)
+	assert.ErrorIs(t, err, oktetoErrors.ErrCouldNotInferAnyManifest)
+}

